### ISSUE TITLE
X11 role is no longer set

### ISSUE THREAD

**Issue by Knaldgas** [pdfpc/pdfpc#550]
Same problem as in #540, although the solution there is not good (it only works because it matches on the instance name).
I use the role to send presentation to projector and presenter to local screen. This no longer work as "role" is no longer set.

I probably use the "name", *if* it was set when the window is created, which it does not seem to be (seems like the full name is "pdfpc" at the point where awesomewm catches it, in which case it cannot distinguish between the two.

Please, either set the name correctly initially, or include the role initially.
Personally I prefer the "role" property as this is what the property is for.

**Comment by fnevgeny** [comment]
I don't understand how it is different from #540 - in which case the `title` attribute ( = `name` in the WM) apparently, worked. BTW, `title` is set in the very beginning of the window constructor, exactly where `role` used to be set.

**Comment by Knaldgas** [comment]
The solution in #540 uses the following rule selector:
```
    rule_any = { 
        instance = { "pdfpc" }, 
        name = "pdfpc - presentation.*"
    },
```
meaning that it triggers on instance = "pdfpc" or name starts with "pdfpc - presentation". The rule will trigger on any instance of pdfpc due to the first rule. I think the second rule is mostly used to generate heat.

After your response I understood that I must have missed something, and found that I had been on a wild goose chase first time around (for some reason I did not pick up on the special meaning of the dash, sorry ...).

My *working* rule is this:
```
    { rule = { class = "Pdfpc" },
      callback = function(c)
         if string.find (c.name, "pdfpc %- presentation") == 1 then
             awful.client.movetoscreen(c, screen:count())
         else
             awful.client.movetoscreen(c, 1)
         end
      end },
```
It triggers on all "Pdfpc" class windows, installs a callback to be called when the window is ready, which in turn places the window on the right monitor.

In short, I have it working again now (thanks!), but I still believe pdfpc should set the role property.
"Presenter" and "Presentation" is the role of pdfpc, so why not set the "role" property, it is exactly why it's there (I think).

**Comment by fnevgeny** [comment]
> so why not set the "role" property, it is exactly why it's there (I think).

As explained in the link in my response to #540, "role" is specific to X11, and is absent in other Gtk backends (e.g., macOS/quartz). Furthermore, it is already dropped from Gtk anyway - for exactly the same reason. <https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/674>.

In short, since there is a way to solve the problem using the OS-independent standard means (as your last comment proves), I see no compelling reason to use an obscure workaround.
